lighthouse-core/report/html/renderer/performance-category-renderer.js

```diff
@@ -301,66 +301,63 @@ class PerformanceCategoryRenderer extends CategoryRenderer {
+    textEl.textContent = Util.i18n.strings.showRelevantAudits;
 
     const filterChoices = /** @type {LH.ReportResult.AuditRef[]} */ ([
       ({acronym: 'All'}),
       ...filterableMetrics,
     ]);
     for (const metric of filterChoices) {
-      // The radio elements are appended into `categoryEl` to allow the sweet ~ selectors to work
       const elemId = `metric-${metric.acronym}`;
-      const radioEl = this.dom.createChildOf(categoryEl, 'input', 'lh-metricfilter__radio', {
-        type: 'radio',
-        name: 'metricsfilter',
-        id: elemId,
-        hidden: 'true',
-      });
       const labelEl = this.dom.createChildOf(metricFilterEl, 'label', 'lh-metricfilter__label', {
         for: elemId,
         title: metric.result && metric.result.title,
       });
       labelEl.textContent = metric.acronym || metric.id;
+      const radioEl = this.dom.createChildOf(labelEl, 'input', 'lh-metricfilter__radio', {
+        type: 'radio',
+        name: 'metricsfilter',
+        id: elemId,
+        hidden: 'true',
+      });
+
       if (metric.acronym === 'All') {
         radioEl.checked = true;
+        labelEl.classList.add('lh-metricfilter__label--active');
       }
-      // Dynamically write some CSS, for the CSS-only filtering
-      labelSelectors.push(`.lh-metricfilter__radio#${elemId}:checked ~ .lh-metricfilter > .lh-metricfilter__label[for="${elemId}"]`); // eslint-disable-line max-len
-      if (metric.relevantAudits) {
-        /* Generate some CSS selectors like this:
-            #metric-CLS:checked ~ .lh-audit-group > #layout-shift-elements,
-            #metric-CLS:checked ~ .lh-audit-group > #non-composited-animations,
-            #metric-CLS:checked ~ .lh-audit-group > #unsized-images
-        */
-        auditSelectors.push(metric.relevantAudits.map(auditId => `#${elemId}:checked ~ .lh-audit-group > #${auditId}`).join(',\n')); // eslint-disable-line max-len
-      }
-    }
+      categoryEl.append(metricFilterEl);
 
-    const styleEl = this.dom.createChildOf(metricFilterEl, 'style');
-    // eslint-disable-next-line max-len
-    styleEl.textContent = `
-${labelSelectors.join(',\n')} {
-  background: var(--color-blue-A700);
-  color: var(--color-white);
-}
-/* If selecting non-All, hide all audits (and also the group header/description… */
-.lh-metricfilter__radio:checked:not(#metric-All) ~ .lh-audit-group .lh-audit,
-.lh-metricfilter__radio:checked:not(#metric-All) ~ .lh-audit-group .lh-audit-group__description,
-.lh-metricfilter__radio:checked:not(#metric-All) ~ .lh-audit-group .lh-audit-group__itemcount {
-  display: none;
-}
-/* …And then display:block the relevant ones */
-${auditSelectors.join(',\n')} {
-  display: block;
-}
-/*# sourceURL=metricfilter.css */
-    `;
-    categoryEl.append(metricFilterEl);
+      // Toggle class/hidden state based on filter choice.
+      radioEl.addEventListener('input', _ => {
+        categoryEl.querySelectorAll('label.lh-metricfilter__label').forEach(elem => {
```

Comment:
nit: you do forEach and a for of loop here. pick one? I suggest for of. using forEach is awkward imo and unless you need the extra args